server/internal/core/application/covenant.go

```diff
@@ -377,19 +383,61 @@ func (s *covenantService) SubmitRedeemTx(context.Context, string) (string, strin
 }
 
 func (s *covenantService) SignVtxos(ctx context.Context, forfeitTxs []string) error {
-	return s.forfeitTxs.sign(forfeitTxs)
+	s.currentRoundLock.Lock()
+	defer s.currentRoundLock.Unlock()
+	currentRound := s.currentRound
+
+	if err := s.forfeitTxs.sign(forfeitTxs); err != nil {
+		return err
+	}
+
+	return s.checkForfeitsAndBoardingSigsSent(currentRound)
```

Comment:
Not sure about this, but maybe better to run this check in a go rountine in order to avoid returning the error to user in case something went wrong there?

Reply:
Done